mypy/checker.py

```diff
@@ -228,6 +229,9 @@
 
 DEFAULT_LAST_PASS: Final = 1  # Pass numbers start at 0
 
+# Maximum length of fixed tuple types inferred when narrowing from variadic tuples.
+MAX_PRECISE_TUPLE_SIZE: Final = 15
```

Comment:
This feels a bit high to me, but I don't have a strong opinion.

Reply:
8 is another option that I was thinking about.